lms/djangoapps/django_comment_client/base/tests.py

```diff
@@ -1395,33 +1416,64 @@ def setUpClass(cls):
     @classmethod
     def setUpTestData(cls):
         super(TeamsPermissionsTestCase, cls).setUpTestData()
-
+        cls.course = CourseFactory.create()
         cls.password = "test password"
         seed_permissions_roles(cls.course.id)
 
-        # Create 3 users-- student in team, student not in team, discussion moderator
-        cls.student_in_team = UserFactory.create(password=cls.password)
-        cls.student_not_in_team = UserFactory.create(password=cls.password)
-        cls.moderator = UserFactory.create(password=cls.password)
-        CourseEnrollmentFactory(user=cls.student_in_team, course_id=cls.course.id)
-        CourseEnrollmentFactory(user=cls.student_not_in_team, course_id=cls.course.id)
-        CourseEnrollmentFactory(user=cls.moderator, course_id=cls.course.id)
+        # Create enrollment tracks
+        verified_coursemode = CourseModeFactory.create(
+            course_id=cls.course.id,
+            mode_slug=CourseMode.VERIFIED
+        )
+        audit_coursemode = CourseModeFactory.create(
+            course_id=cls.course.id,
+            mode_slug=CourseMode.AUDIT
+        )
+
+        # Create 6 users--
+        # student in team (in the team, audit)
+        # student not in team (not in the team, audit)
+        # cohorted (in the cohort, audit)
+        # verified (not in the cohort, verified)
+        # moderator (in the cohort, audit, moderator permissions)
+        # group moderator (in the cohort, verified, group moderator permissions)
+        def create_users_and_enroll(coursemode):
+            student = UserFactory.create(password=cls.password)
+            CourseEnrollmentFactory(
+                course_id=cls.course.id,
+                user=student,
+                mode=coursemode
+            )
+            return student
+
+        cls.student_in_team, cls.student_not_in_team, cls.moderator, cls.cohorted = (
+            [create_users_and_enroll(audit_coursemode) for i in range(4)])
+        cls.verified, cls.group_moderator = [create_users_and_enroll(verified_coursemode) for i in range(2)]
+
+        # Give moderator and group moderator permissions
         cls.moderator.roles.add(Role.objects.get(name="Moderator", course_id=cls.course.id))
+        assign_role(cls.course.id, cls.group_moderator, 'Group Moderator')
```

Comment:
Should you be assigning the role the same way the Moderator role is set above on line 1454?